bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -682,10 +683,31 @@ private void onResource(Resource resource, boolean updateChannels) {
                 updateDependenciesTask = scheduler.submit(() -> updateDependencies());
             }
         } else {
+            if (SUPPORTED_SCENE_TYPES.contains(resource.getType())) {
+                switch (resource.getContentType()) {
+                    case ADD:
+                        if (getResourceReference().equals(resource.getGroup())) {
+                            resourceConsumed = true;
+                            resource.setContentType(ContentType.FULL_STATE);
+                            sceneContributorsCache.put(resource.getId(), resource);
```

Comment:
Optional, just feedback/ideas. Consider:
```suggestion
                            putResourceToCache(resource);
```

for consistency, i.e. to use the same logic.

Also I'm wondering if the method should return here. In lines 707-715 we are reading from the cache what we just put there, and then updating it once more. The only thing missing would be the logging statement in lines 717-719. It's not performance I'm worried about, but reading the code is a bit hard because you have to carefully read through it all to understand that it is doing the same job twice, but in a (currently) harmless way.

Perhaps all logic related to scenes could be extracted to a separate method (e.g. `onSceneResource`). Then we could even drop the methods `putResourceToCache` and `getResourceFromCache` since there would be two distinct paths with different handling and different caches.

Reply:
Moved the scene add delete processing to a separate method..